pkg/streamingpromql/engine.go

```diff
@@ -44,12 +47,28 @@ func NewEngine(opts EngineOpts, limitsProvider QueryLimitsProvider, metrics *sta
 		return nil, errors.New("enabling delayed name removal not supported by Mimir query engine")
 	}
 
+	// Sort DisabledFunctions to optimise lookups
+	sort.Strings(opts.Features.DisabledFunctions)
```

Comment:
`slices.Sort` is preferred as of Go 1.22 (`sort.Strings` just calls `slices.Sort` now)